Clicking "More info" on Play integrity API usage notification for an uninstalled app crashes settings

I wanted to try the new Play Integrity API notification, so I installed Authy. The notification popped up as expected, but since the notification persists even after app uninstall, I'm able to click "More info" even with the app uninstalled, which consistently crashes the settings app.

Backtrace 1:
```
type: crash
osVersion: google/husky/husky:15/AP4A.250105.002/2025012600:user/release-keys
flags: dev options enabled
package: com.android.settings:35, targetSdk 35
sharedUid: android.uid.system
process: com.android.settings
processUptime: 257 + 212 ms

java.lang.RuntimeException: Unable to resume activity {com.android.settings/com.android.settings.Settings$AppManagePlayIntegrityApiActivity}: java.lang.NullPointerException: Attempt to read from field 'android.content.pm.ApplicationInfo android.content.pm.PackageInfo.applicationInfo' on a null object reference in method 'boolean com.android.settings.applications.AppManagePlayIntegrityApiFragment.refreshUi()'
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5424)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5457)
	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:73)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:63)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleItem(TransactionExecutor.java:169)
	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:101)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:80)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2728)
	at android.os.Handler.dispatchMessage(Handler.java:109)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8826)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
Caused by: java.lang.NullPointerException: Attempt to read from field 'android.content.pm.ApplicationInfo android.content.pm.PackageInfo.applicationInfo' on a null object reference in method 'boolean com.android.settings.applications.AppManagePlayIntegrityApiFragment.refreshUi()'
	at com.android.settings.applications.AppManagePlayIntegrityApiFragment.refreshUi(AppManagePlayIntegrityApi.kt:111)
	at com.android.settings.applications.AppInfoBase.onResume(AppInfoBase.java:117)
	at androidx.fragment.app.Fragment.performResume(Fragment.java:3210)
	at androidx.fragment.app.FragmentStateManager.resume(FragmentStateManager.java:666)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:310)
	at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:114)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1684)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3316)
	at androidx.fragment.app.FragmentManager.dispatchResume(FragmentManager.java:3248)
	at androidx.fragment.app.FragmentController.dispatchResume(FragmentController.java:285)
	at androidx.fragment.app.FragmentActivity.onResumeFragments(FragmentActivity.java:333)
	at androidx.fragment.app.FragmentActivity.onPostResume(FragmentActivity.java:322)
	at android.app.Activity.performResume(Activity.java:9257)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5414)
	... 16 more
```

Backtrace 2:
```
type: crash
osVersion: google/husky/husky:15/AP4A.250105.002/2025012600:user/release-keys
flags: dev options enabled
package: com.android.settings:35, targetSdk 35
sharedUid: android.uid.system
process: com.android.settings
processUptime: 201 + 233 ms

java.lang.RuntimeException: Unable to resume activity {com.android.settings/com.android.settings.Settings$AppManagePlayIntegrityApiActivity}: java.lang.NullPointerException: Attempt to read from field 'android.content.pm.ApplicationInfo android.content.pm.PackageInfo.applicationInfo' on a null object reference in method 'boolean com.android.settings.applications.AppManagePlayIntegrityApiFragment.refreshUi()'
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5424)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5457)
	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:73)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:63)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleItem(TransactionExecutor.java:169)
	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:101)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:80)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2728)
	at android.os.Handler.dispatchMessage(Handler.java:109)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8826)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
Caused by: java.lang.NullPointerException: Attempt to read from field 'android.content.pm.ApplicationInfo android.content.pm.PackageInfo.applicationInfo' on a null object reference in method 'boolean com.android.settings.applications.AppManagePlayIntegrityApiFragment.refreshUi()'
	at com.android.settings.applications.AppManagePlayIntegrityApiFragment.refreshUi(AppManagePlayIntegrityApi.kt:111)
	at com.android.settings.applications.AppInfoBase.onResume(AppInfoBase.java:117)
	at androidx.fragment.app.Fragment.performResume(Fragment.java:3210)
	at androidx.fragment.app.FragmentStateManager.resume(FragmentStateManager.java:666)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:310)
	at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:114)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1684)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3316)
	at androidx.fragment.app.FragmentManager.dispatchResume(FragmentManager.java:3248)
	at androidx.fragment.app.FragmentController.dispatchResume(FragmentController.java:285)
	at androidx.fragment.app.FragmentActivity.onResumeFragments(FragmentActivity.java:333)
	at androidx.fragment.app.FragmentActivity.onPostResume(FragmentActivity.java:322)
	at android.app.Activity.performResume(Activity.java:9257)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5414)
	... 16 more
```

--- a/src/com/android/settings/applications/AppManagePlayIntegrityApi.kt
+++ b/src/com/android/settings/applications/AppManagePlayIntegrityApi.kt
@@ -103,26 +103,32 @@ class AppManagePlayIntegrityApiFragment : AppInfoWithHeader() {
     }
 
     override fun refreshUi(): Boolean {
+        val appInfo = getAppInfo(mPackageName) ?: return false
         val gosPs = GosPackageState.get(mPackageName, mUserId)
 
         showUsageNotifsSwitch.isChecked = AswBlockPlayIntegrityApi.I.isNotificationEnabled(gosPs)
 
         blockUsageAttemptsSwitch.isChecked = AswBlockPlayIntegrityApi.I
-                .get(requireContext(), mUserId, mPackageInfo.applicationInfo!!, gosPs)
+                .get(requireContext(), mUserId, appInfo, gosPs)
 
         return true
     }
 
     private fun isPlayStoreAvailable(): Boolean {
-        val pkgManager = requireContext().packageManager
-        val appInfo = try {
-            pkgManager.getApplicationInfoAsUser(PackageId.PLAY_STORE_NAME, 0, mUserId)
-        } catch (e: NameNotFoundException) { return false }
-
+        val appInfo = getAppInfo(PackageId.PLAY_STORE_NAME) ?: return false
         if (!appInfo.enabled) {
             return false
         }
         return appInfo.ext().packageId == PackageId.PLAY_STORE
+    }
+
+    private fun getAppInfo(pkgName: String): ApplicationInfo? {
+        val pkgManager = requireContext().packageManager
+        try {
+            return pkgManager.getApplicationInfoAsUser(pkgName, 0, mUserId)
+        } catch (e: NameNotFoundException) {
+            return null
+        }
     }
 
     override fun createDialog(id: Int, errorCode: Int): AlertDialog? = null

